### PR TITLE
Move frontend functionalities out of Tagging and put them into Vanilla app

### PR DESCRIPTION
**Pull request by DaazKu**
Fixes https://github.com/vanilla/vanilla/issues/3970

No functionality changes!
Removes dead code/assets, move hooks and module into core.

Second part (where we will remove the plugin entirely) will be addressed by https://github.com/vanilla/vanilla/issues/3971

### CODE REVIEW

File: applications/vanilla/settings/class.hooks.php [hunk 1]
```diff
@@ -153,13 +154,246 @@ public function deleteUserData($userID, $options = array(), &$data = null) {
...
      */
     public function discussionController_beforeCommentBody_handler($sender) {
         Gdn::regarding()->beforeCommentBody($sender);
     }
 
     /**
+     *
+     */
+    public function discussionController_afterDiscussionBody_handler($Sender) {
+        /* Show tags after discussion body. */
```
Comment (linc) [inline applications/vanilla/settings/class.hooks.php]: Why is the docblock comment in here?

File: applications/vanilla/controllers/class.categoriescontroller.php [hunk 2]
```diff
@@ -455,6 +456,7 @@ public function all($Category = '', $displayAs = '') {
         $this->addModule('DiscussionFilterModule');
         $this->addModule('BookmarkedModule');
         $this->addModule($CategoryFollowToggleModule);
+        $this->addModule('TagModule');
```
Comment (linc) [inline applications/vanilla/controllers/class.categoriescontroller.php]: I'm not a fan of this module creeping in everywhere by default. Can we keep this part in the plugin?

I think the line is we want all the ability to *use* tagging in core, but turning on "users adding / using arbitrary tags" stays in the plugin.

File: applications/vanilla/controllers/class.discussionscontroller.php [hunk 3]
```diff
@@ -675,4 +681,132 @@ public function promoted() {
         Gdn_Theme::section('PromotedContent');
         $this->render('promoted', 'modules', 'vanilla');
     }
+
+    /**
+     *
```
Comment (linc) [inline applications/vanilla/controllers/class.discussionscontroller.php]: :(

File: applications/vanilla/settings/class.hooks.php [hunk 4]
```diff
@@ -153,13 +154,246 @@ public function deleteUserData($userID, $options = array(), &$data = null) {
     /**
      *
```
Comment (linc) [inline applications/vanilla/settings/class.hooks.php]: :(

File: applications/vanilla/settings/class.hooks.php [hunk 5]
```diff
@@ -153,13 +154,246 @@ public function deleteUserData($userID, $options = array(), &$data = null) {
...
+            $tags = TagModel::instance()->getDiscussionTags($discussionID, TagModel::IX_EXTENDED);
+
+            foreach ($tags as $key => $value) {
+                $sender->setData('Discussion.'.$key, $value);
+            }
+        }
+    }
+
+    /**
+     *
```
Comment (linc) [inline applications/vanilla/settings/class.hooks.php]: :(

File: applications/vanilla/settings/class.hooks.php [hunk 6]
```diff
@@ -153,13 +154,246 @@ public function deleteUserData($userID, $options = array(), &$data = null) {
...
+            }
+            if (count($Tags) > $maxTags) {
+                $Sender->Validation->addValidationResult('Tags', '@'.sprintf(t('You can only specify up to %s tags.'), $maxTags));
+            }
+        }
+    }
+
+    /**
+     * Save tags when saving a discussion.
+     */
```
Comment (linc) [inline applications/vanilla/settings/class.hooks.php]: @param

**Comment by linc** [comment]
When I deploy this change, no one's sites should fundamentally change. If they have "Tagging" on, everyone can keep tagging. If they don't have "Tagging" on, plugins should still be able to use all of its functionality a la Reactions. But it shouldn't expose new features in the New Discussion form now all of a sudden. That's still opt-in functionality.

**Comment by DaazKu** [comment]
I have another task that says to move tag administration into core: https://github.com/vanilla/vanilla/issues/3971

When I spoke with Todd my understanding of this task was that I need to move everything into vanilla and that is what I did.
In the second task I will move the rest (everything that is administration related) into core and add a new config to display UI components into vanilla. It will need to be backward compatible.

> When I deploy this change, no one's sites should fundamentally change

The plugin still needs to be enabled or the UI components won't be rendered.

File: applications/vanilla/controllers/class.categoriescontroller.php (hunk 2 above)
Comment (DaazKu) [inline applications/vanilla/controllers/class.categoriescontroller.php]: The module does nothing is the plugin is not enabled. See https://github.com/vanilla/vanilla/pull/5049#issuecomment-275528636

**Comment by DaazKu** [comment]
I'll update the docblocks properly!

File: applications/vanilla/settings/class.hooks.php [hunk 7]
```diff
@@ -153,13 +154,246 @@ public function deleteUserData($userID, $options = array(), &$data = null) {
...
+
+    /**
+     * Add the tag input to the discussion form.
+     *
+     * @param Gdn_Controller $Sender
+     */
+    public function postController_afterDiscussionFormOptions_handler($Sender) {
+//        if (!c('EnabledPlugins.Tagging')) {
+//            return;
+//        }
```
Comment (linc) [inline applications/vanilla/settings/class.hooks.php]: Forget to uncomment this?

Comment (DaazKu) [inline applications/vanilla/settings/class.hooks.php]: yup.....

**Pull request merged**